numeric.rb

```diff
@@ -148,3 +148,60 @@ def zero?
     Primitive.cexpr! 'rb_int_zero_p(self)'
   end
 end
+
+class Float
+  #
+  # call-seq:
+  #    float.to_f  ->  self
+  #
+  # Since +float+ is already a Float, returns +self+.
+  #
+  def to_f
+    return self
+  end
+
+  #
+  #  call-seq:
+  #     float.abs        ->  float
+  #     float.magnitude  ->  float
+  #
+  #  Returns the absolute value of +float+.
+  #
+  #     (-34.56).abs   #=> 34.56
+  #     -34.56.abs     #=> 34.56
+  #     34.56.abs      #=> 34.56
+  #
+  #  Float#magnitude is an alias for Float#abs.
+  #
+  def abs
+    Primitive.attr! 'inline'
+    Primitive.cexpr! 'rb_float_abs(self)'
+  end
+
+  def magnitude
+    Primitive.attr! 'inline'
+    Primitive.cexpr! 'rb_float_abs(self)'
+  end
+
+  #
+  # call-seq:
+  #    -float  ->  float
+  #
+  # Returns +float+, negated.
+  #
+  def -@
+    Primitive.attr! 'inline'
+    Primitive.cexpr! 'rb_float_uminus(self)'
```

Comment:
To understand better, does using `Primitive.attr! 'inline'` require using `Primitive.cexpr!`?
Or could it also be done like:
```ruby
  def -@
    Primitive.attr! 'inline'
    Primitive.rb_float_uminus
  end
```
?
(which seems nicer to me because it's not mixing C code inside Ruby code)

Reply:
It could be. To use that, we need to have ec as the first argument even if it's not needed, which is again ko1's design. (I don't like this particular part because I don't like taking unused arguments. Therefore cexpr is sometimes better)

Reply:
I think an unused `ec` argument is fine, and would be cleaner.
cc @ko1

Another advantage of using a named Primitive and not `cexpr!` is the Ruby code could be reused on other Ruby implementations. For `Float#-@` it doesn't matter much, but as more methods are declared like this it will become valuable.

Actually, I just noticed a difference, on TruffleRuby, `Primitive.foo` pass no arguments (`Primitive.rb_float_uminus(self)` would be needed), but on CRuby, it seems `self` is always passed "implicitly".